test/unit/models/flag.test.js

```diff
@@ -0,0 +1,46 @@
+const chai = require("chai");
+const { expect } = chai;
+const chaiHttp = require("chai-http");
+
+const addUser = require("../../utils/addUser");
+const cleanDb = require("../../utils/cleanDb");
+
+const flagModel = require("../../../models/flag");
+const firestore = require("../../../utils/firestore");
+const flagFirestore = firestore.collection("featureFlags");
+
+const flagData = {
+  title: "dark - mode",
+  enabled: true,
+  roles: ["members", "no_role", "super_user"],
+  users: {
+    "pallab id": true,
+    "rohit id": false,
+  },
+};
+
+chai.use(chaiHttp);
+
+describe("flags", function () {
+  beforeEach(async function () {
+    await addUser();
+  });
+
+  afterEach(async function () {
+    await cleanDb();
+  });
+
+  describe("add feature flag", function () {
+    it("Should return flag Id", async function () {
+      const flagId = await flagModel.addFlag(flagData);
+      const data = (await flagFirestore.doc(flagId).get()).data();
+      Object.keys(flagData).forEach((key) => {
+        expect(flagData[key]).to.deep.equal(data[key]);
+      });
+      expect(data.title).to.be.a("string");
+      expect(data.enabled).to.be.a("boolean");
+      expect(data.roles).to.be.a("array");
+      expect(data.users).to.be.a("object");
```

Comment:
Type comparison is not needed as it's already being covered when comparing mockdata directly. Can we compare flagId instead? 

```js
      const flagId = await flagModel.addFlag(flagData);
      const flagDoc = await flagFirestore.doc(flagId).get();

      const dataId = flagDoc.id;
      const data = flagDoc.data();
      
      // compare dataId and flagId
```